src/_P017_PN532.ino

```diff
@@ -112,7 +112,9 @@ boolean Plugin_017(byte function, struct EventStruct *event, String& string)
         counter++;
         if (counter == 3 )
         {
-          if (digitalRead(4) == 0 || digitalRead(5) == 0)
+          // TODO: is it needed?
+          if (Settings.Pin_i2c_sda >= 0 && Settings.Pin_i2c_scl>= 0 
+              && (digitalRead(Settings.Pin_i2c_sda)==0 || digitalRead(Settings.Pin_i2c_scl)==0))
```

Comment:
It looks like some extra check for clock stretching that is part of the I2C standard.
So I would say it should not be part of the plugin, but of the Wire class.

Or maybe it is a state signalling of this specific chip to indicate it is crashed? (would be a strange way, to block the entire I2C bus)

Reply:
In fact there are some situation when pn532 uses stretching, will check datasheet+manual from NXP

Reply:
@TD-er found reported here: https://github.com/esp8266/Arduino/issues/1541

Reply:
> @TD-er found reported here: [esp8266/Arduino#1541](https://github.com/esp8266/Arduino/issues/1541)

OK, so better leave it in then.

Reply:
ok comment updated